gnovm/pkg/repl/repl.go

```diff
@@ -0,0 +1,270 @@
+package repl
+
+import (
+	"bytes"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"text/template"
+
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/gnovm/tests"
+)
+
+const (
+	executedFunc = "main"
+	fileTemplate = `// generated by 'gno repl'
+// FILE NAME: test{{$.ID}}.gno
+package test
+{{range .Imports}}
+{{.}}
+{{end}}
+{{range .Declarations}}
+{{.}}
+{{end}}
+{{with .Expression}}
+// main function
+func main{{$.ID}}() {
+	{{.}}
+}
+{{end}}
+`
+)
+
+type tempModel struct {
+	ID           int
+	Declarations []string
+	Imports      []string
+	Expression   string
+}
+
+type state struct {
+	// b is the buffer where the VM is writing. It is used to get the output when a expression is executed
+	b bytes.Buffer
+
+	// imports contains all the imports added. they will be added to the following generated source code
+	imports map[string]string
+
+	// files are all generated files. This is used when source code is requested
+	files map[string]string
+
+	// id is the actual execution number. Used to avoid duplicated functions and file names.
+	id int
+
+	fileset *token.FileSet
+	machine *gno.Machine
+}
+
+type Repl struct {
+	state *state
+	tmpl  *template.Template
+}
+
+func newState() *state {
+	s := &state{
+		imports: make(map[string]string),
+		files:   make(map[string]string),
+		fileset: token.NewFileSet(),
+	}
+
+	s.machine = newMachine(&s.b)
+
+	return s
+}
+
+func newMachine(b *bytes.Buffer) *gno.Machine {
+	testStore := tests.TestStore("teststore", "", b, b, b, tests.ImportModeStdlibsOnly)
+	return gno.NewMachineWithOptions(gno.MachineOptions{
+		PkgPath: "test",
+		Output:  b,
+		Store:   testStore,
+	})
+}
+
+// NewRepl creates a Repl struct. It is able to process input source code and eventually run it.
+func NewRepl() *Repl {
+	t, err := template.New("tmpl").Parse(fileTemplate)
+	if err != nil {
+		panic(fmt.Errorf("error parsing template: %w", err))
+	}
```

Comment:
```suggestion
	t := template.Must(template.New("tmpl").Parse(fileTemplate))
```

Reply:
thanks for pointing it out, I didn't know that helper!

Reply:
Addressed here: [cf02e10](https://github.com/gnolang/gno/pull/978/commits/cf02e1028ec4ff2109a7b9ae12464008f98ca1bc)